Code/Tools/AzTestRunner/src/main.cpp

```diff
@@ -163,12 +198,20 @@ namespace AzTestRunner
         // run the test main function.
         if (testMainFunction->IsValid())
         {
-            result = (*testMainFunction)(argc, argv);
+            // collapse the arguments vector into a c-style array of character pointers
+            std::vector<char*> cArguments;
+            cArguments.reserve(arguments.size());
+            for (const auto& argument : arguments)
+            {
+                cArguments.push_back(const_cast<char*>(argument.c_str()));
+            }
+
+            result = (*testMainFunction)(static_cast<int>(cArguments.size()), cArguments.data());
```

Comment:
The `const_cast` can be deferred until the `testMainFunction` pointer is invoke.
This means that an `AZStd::fixed_vector<const char*>` can be used instead.

Also the `cArguments` doesn't fit within the coding standard.
```suggestion
            // collapse the arguments vector into a c-style array of character pointers
            AZStd::fixed_vector<const char*> charArguments(;
            charArguments.reserve(argStorage.size());
            for (AZStd::string_view argument : argStorage)
            {
                charArguments.push_back(argument.data());
            }

            result = (*testMainFunction)(static_cast<int>(charArguments.size()), const_cast<char*>(charArguments.data()));
```

Reply:
The number of args is not known at compile time so I couldn't use the fixed vector but I deferred the const cast and renamed the char vec👍